cypress/e2e/shared/notificationRules.test.ts

```diff
@@ -292,4 +302,457 @@ describe('NotificationRules', () => {
+                source: 'wumpus',
+                sent: 'true',
+                valField: {key: 'foo', val: 'bar'},
+                sourceTimestamp: calcNanoTimestamp('17m'),
+                statusTimestamp: calcNanoTimestamp('16m'),
+                message: 'This is a fake critical message',
+              },
+              '15m'
+            )
+
+            writeMonitoringRecord(
+              {
+                check: check,
+                endp: endp,
+                rule: rule,
+                level: 'info',
+                source: 'wumpus',
+                sent: 'false',
+                valField: {key: 'foo', val: 'bar'},
+                sourceTimestamp: calcNanoTimestamp('12m'),
+                statusTimestamp: calcNanoTimestamp('11m'),
+                message: 'This is a fake info message',
+              },
+              '10m'
+            )
+            writeMonitoringRecord(
+              {
+                check: check,
+                endp: endp,
+                rule: rule,
+                level: 'warn',
+                source: 'wumpus',
+                sent: 'true',
+                valField: {key: 'foo', val: 'bar'},
+                sourceTimestamp: calcNanoTimestamp('22m'),
+                statusTimestamp: calcNanoTimestamp('21m'),
+                message: 'This is a fake warning message',
+              },
+              '20m'
+            )
+            writeMonitoringRecord(
+              {
+                check: check,
+                endp: endp,
+                rule: rule,
+                level: 'ok',
+                source: 'wumpus',
+                sent: 'true',
+                valField: {key: 'foo', val: 'bar'},
+                sourceTimestamp: calcNanoTimestamp('7m'),
+                statusTimestamp: calcNanoTimestamp('6m'),
+                message: 'This is a fake OK message',
+              },
+              '5m'
+            )
+          })
+        })
+      })
+
+      // use direct route to page
+      cy.get<Organization>('@org').then(({id}: Organization) => {
+        cy.get<NotificationRule>('@rule').then(rule => {
+          cy.fixture('routes').then(({orgs}) => {
+            cy.visit(
+              `${orgs}/${id}/alert-history?type=notifications&filter="notificationRuleID%20%3D%3D%20%22${rule.id}"`
+            )
+            cy.url().should('include', `${orgs}/${id}/alert-history`)
+          })
+        })
+      })
+    })
+
+    it('shows and filters history', () => {
+      cy.get('.event-row').should('have.length', 4)
+      cy.get('.event-row--field:nth-of-type(1)').then(timestamps => {
+        const stamps = timestamps
+          .toArray()
+          .map(n => new Date(n.innerText).getTime())
+        // assert sort desc by date
+        for (let i = 1; i < stamps.length; i++) {
+          expect(stamps[i - 1]).to.be.above(stamps[i])
+        }
+      })
+      // verify first row
+      cy.get('[class$=ScrollContainer] > div:nth-of-type(1)').within(() => {
```

Comment:
is there a reason we aren't using testIDs here? I find being dependent on CSS class names to be an issue because those change pretty often. We're working on a pretty major overhaul of Clockface which will probably change most CSS class names and this test will have to be completely re-written. If we use testIDs, we don't have that issue

Reply:
I agree, data-testid values are preferable because theoretically less prone to change.  However, when investigating the Check Statuses page, I could find no `data-testid` attributes below `data-testid='page-contents'`.  Coming from QA, I'm hesitant to modify code in the application under test, in other words to insert `data-testid` attributes into the page.  However, if you have no problem with this, I'll go ahead and do it and then modify the test code accordingly.

Reply:
Yeah I think we should add the testID attributes if you don't mind. Most of the Clockface components come with generic testIDs for free. But if one is missing, adding it to the component here in the UI repo is the right move. Sorry to suggest so much refactoring...but I know this Clockface overhaul is coming soon and a lot of these CSS classes will change so it's important that we use testIDs for new tests